### PR TITLE
test: Remove trivial `ShapedModule`s

### PR DESCRIPTION
**Pull request by ValerianRey**
* Add support for RNN, BatchNorm2d and InstanceNorm2d in get_in_out_shapes
* Remove WithRNN, WithBatchNorm and WithModuleTrackingRunningStats - use simple factories instead

Whenever a ShapedModule is simply a wrapper around a single nn.Module, we can replace it by a line in get_in_out_shapes and make a factory directly out of the nn.Module. It's not strictly equivalent (because now the module isn't wrapped), but it's closer to what a user would do.

### CODE REVIEW

**Comment by ValerianRey** [comment]
@PierreQuinton FYI, this is how you can test any nn.Module now.
